packages/wrapped-keys/src/lib/lit-actions-client/batch-generate-keys.ts

```diff
@@ -0,0 +1,56 @@
+import { GLOBAL_OVERWRITE_IPFS_CODE_BY_NETWORK } from '@lit-protocol/constants';
+import { AccessControlConditions } from '@lit-protocol/types';
+
+import { postLitActionValidation } from './utils';
+import { BatchGeneratePrivateKeysParams, Network } from '../types';
+
+interface BatchGeneratePrivateKeysWithLitActionParams
+  extends BatchGeneratePrivateKeysParams {
+  accessControlConditions: AccessControlConditions;
+  litActionIpfsCid?: string;
+  litActionCode?: string;
+}
+
+interface GeneratePrivateKeyLitActionResult {
+  ciphertext: string;
+  dataToEncryptHash: string;
+  publicKey: string;
+  memo: string;
+}
+
+interface BatchGeneratePrivateKeysWithLitActionResult {
+  network: Network;
+  signedMessage?: { signature: string };
+  generatedPrivateKey: GeneratePrivateKeyLitActionResult;
+}
+
+export async function batchGenerateKeysWithLitAction(
+  args: BatchGeneratePrivateKeysWithLitActionParams
+): Promise<BatchGeneratePrivateKeysWithLitActionResult[]> {
+  const {
+    accessControlConditions,
+    litNodeClient,
+    actions,
+    pkpSessionSigs,
+    litActionIpfsCid,
+    litActionCode,
+  } = args;
+
+  const result = await litNodeClient.executeJs({
+    numResponsesRequired: 1,
```

Comment:
This sends to only one node or sends to all but waits on only one? @MaximusHaximus

Reply:
Sends to all, but returns the _first successful response_.

This way...
1. We aren't unlucky and waiting on a node that we chose randomly that is slow
2. If _any_ nodes return a successful response, the request succeeds, instead of the request failing if the one random node we chose throws an error.

Reply:
I think we should make this change as in our experience the nodes returns the value if they're hit, so we're reasonably confident about it and it reduce the load on the network as well @MaximusHaximus

Reply:
All set!